### PR TITLE
 Reader Conversations: use compact card + hook up comments

### PR DESCRIPTION
**Pull request by samouri**
The reader conversations tool should use the compact cards we've designed specifically for it.

screen shot 2017-07-26 at 1 45 44 pm

Note: this exposes css tweaks needed.

### CODE REVIEW

File: client/blocks/reader-post-card/compact.jsx [hunk 1]
```diff
@@ -11,45 +13,61 @@ import Emojify from 'components/emojify';
...
 import FeaturedAsset from './featured-asset';
+import { getDateSortedPostComments } from 'state/comments/selectors';
+import ConversationsComments from 'blocks/conversations/list';
 
-const CompactPost = ( { post, postByline, children, isDiscover } ) => {
+const CompactPost = ( { post, postByline, children, isDiscover, comments } ) => {
 	/* eslint-disable wpcalypso/jsx-classname-namespace */
+	const commentIdsToShow = map( take( comments, 3 ), 'ID' );
 	return (
-		<div className="reader-post-card__post">
```
Comment (samouri) [inline client/blocks/reader-post-card/compact.jsx]: just added a single level of indentation here to support the new div

File: client/blocks/reader-post-card/compact.jsx [hunk 2]
```diff
@@ -11,45 +13,61 @@ import Emojify from 'components/emojify';
...
+					</AutoDirection>
+					<ReaderExcerpt post={ post } isDiscover={ isDiscover } />
+					{ children }
+				</div>
 			</div>
+			<ConversationsComments
+				post={ post }
+				postId={ post.ID }
+				blogId={ post.site_ID }
+				commentIds={ commentIdsToShow }
```
Comment (blowery) [inline client/blocks/reader-post-card/compact.jsx]: we can just get this set off the post key

**Comment by blowery** [comment]
i took a slightly different approach in https://github.com/Automattic/wp-calypso/pull/16421/files#diff-1522e7916e40bbd2c8a2e5c27e55f80b

that way we can just use the comment id list on the post key and not make a "compact" card contain comments or the extra div.

File: client/blocks/reader-post-card/compact.jsx (hunk 2 above)
Comment (samouri) [inline client/blocks/reader-post-card/compact.jsx]: do you think thats better? its definitely cheaper

**Comment by samouri** [comment]
> that way we can just use the comment id list on the post key and not make a "compact" card contain comments or the extra div.

is that in case we ever want to use either component separately? 
I like it -- will need to think of a good home for conversationpost.  I don't like post-lifecycle

**Comment by blowery** [comment]
> I like it -- will need to think of a good home for conversationpost. I don't like post-lifecycle

In a former life, the `Stream` component used to take a `cardFactory` which took a `postKey` and returned the class to use for that particular key. Could go back to something like that, but ... `PostLifecycle` really took over that role.

File: client/blocks/reader-post-card/compact.jsx (hunk 2 above)
Comment (blowery) [inline client/blocks/reader-post-card/compact.jsx]: redux isn't the truth though

Comment (blowery) [inline client/blocks/reader-post-card/compact.jsx]: that said, this seems fine and is likely easier to deal with down the road...

**Pull request merged**